lib/records/RecCore.cc

```diff
@@ -1145,13 +1145,17 @@ RecConfigReadConfigDir()
 //-------------------------------------------------------------------------
 // RecConfigReadRuntimeDir
 //-------------------------------------------------------------------------
+// ENV variable is the way to overwrite proxy.config.local_state_dir of records.config
 std::string
 RecConfigReadRuntimeDir()
 {
-  char buf[PATH_NAME_MAX];
+  char buf[PATH_NAME_MAX] = {0};
 
-  buf[0] = '\0';
-  RecGetRecordString("proxy.config.local_state_dir", buf, PATH_NAME_MAX);
+  if (const char *env = getenv("PROXY_CONFIG_LOCAL_STATE_DIR")) {
```

Comment:
None of these environment variable overrides are needed since the core already accepts environment variables at higher precedence than configuration value (see `RecConfigOverrideFromEnvironment()`). The comment on `RecConfigReadConfigDir` explains why a special case is needed there.

Reply:
Fixed